src/Microsoft.ApiDesignGuidelines.Analyzers/Core/UriParametersShouldNotBeStrings.Fixer.cs

```diff
@@ -1,30 +1,119 @@
 // Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System;
 using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
 using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
 using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Text;
 
 namespace Microsoft.ApiDesignGuidelines.Analyzers
 {
     /// <summary>
     /// CA1054: Uri parameters should not be strings
     /// </summary>
-    public abstract class UriParametersShouldNotBeStringsFixer : CodeFixProvider
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    public class UriParametersShouldNotBeStringsFixer : CodeFixProvider
     {
         public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(UriParametersShouldNotBeStringsAnalyzer.RuleId);
 
         public sealed override FixAllProvider GetFixAllProvider()
         {
-            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+            // Fixes all occurrences within within Document, Project, or Solution
             return WellKnownFixAllProviders.BatchFixer;
         }
 
         public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            // This is to get rid of warning CS1998, please remove when implementing this analyzer
-            await new Task(() => { });
-            throw new NotImplementedException();
+            var title = MicrosoftApiDesignGuidelinesAnalyzersResources.UriParametersShouldNotBeStringsTitle;
+
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+            var span = context.Span;
+
+            var generator = SyntaxGenerator.GetGenerator(document);
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var parameter = root.FindNode(span, getInnermostNodeForTie: true);
+            if (parameter == null)
+            {
+                // this diagnostic is not something we can deal with
+                return;
+            }
+
+            var methodNode = generator.GetDeclaration(parameter, DeclarationKind.Method);
+            if (methodNode == null)
+            {
+                // this diagnostic is not something we can deal with
+                return;
+            }
+
+            var targetNode = generator.GetDeclaration(parameter, DeclarationKind.Class) ?? generator.GetDeclaration(parameter, DeclarationKind.Struct);
```

Comment:
first time I am using generator and found a bug :) DeclarationKind is missing module (https://github.com/dotnet/roslyn/issues/10094)